compile error about _create_script in ScriptLanguageExtension::_bind_methods

### Tested versions

Tested version: latest version (4.6) from git commit(5ad8b27d8db0776700c2b08bf68fbea5e0a4680d)


### System information

Linux-Godot4.6

### Issue description

Error: 
```
core/object/script_language_extension.cpp: In static member function 'static void ScriptLanguageExtension::_bind_methods()':
./core/object/object.h:386:87: error: '_gdvirtual__create_script_get_method_info' was not declared in this scope; did you mean '_gdvirtual__reload_scripts_get_method_info'?
  386 | #define GDVIRTUAL_BIND(m_name, ...) ::ClassDB::add_virtual_method(get_class_static(), _gdvirtual_##m_name##_get_method_info(), true, sarray(__VA_ARGS__));
      |                                                                                       ^~~~~~~~~~~
core/object/script_language_extension.cpp:107:9: note: in expansion of macro 'GDVIRTUAL_BIND'
  107 |         GDVIRTUAL_BIND(_create_script);
```

Reason:
in source file, GDVIRTUAL_BIND(_create_script) is written in DISABLE_DEPRECATED block
but in header, GDVIRTUAL0RC(Object *, _create_script) is out of DISABLE_DEPRECATED block

Solution:
GDVIRTUAL0RC(Object *, _create_script) should go inside of DISABLE_DEPRECATED block like:
```
#ifndef DISABLE_DEPRECATED
	GDVIRTUAL0RC(Object *, _create_script)
	GDVIRTUAL0RC(bool, _has_named_classes)
#endif
```

instead of  as current version
```
GDVIRTUAL0RC(Object *, _create_script)
#ifndef DISABLE_DEPRECATED
	GDVIRTUAL0RC(bool, _has_named_classes)
#endif
```


### Steps to reproduce

Build command: scons target=editor deprecated=no 

### Minimal reproduction project (MRP)

It is not about godot project but godot source tree itself

--- a/core/object/script_language_extension.cpp
+++ b/core/object/script_language_extension.cpp
@@ -104,8 +104,8 @@ void ScriptLanguageExtension::_bind_methods() {
 	GDVIRTUAL_BIND(_validate, "script", "path", "validate_functions", "validate_errors", "validate_warnings", "validate_safe_lines");
 
 	GDVIRTUAL_BIND(_validate_path, "path");
-	GDVIRTUAL_BIND(_create_script);
 #ifndef DISABLE_DEPRECATED
+	GDVIRTUAL_BIND(_create_script);
 	GDVIRTUAL_BIND(_has_named_classes);
 #endif
 	GDVIRTUAL_BIND(_supports_builtin_mode);

